### PR TITLE
Mark expo upload:ios as unsupported

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

`upload:ios` is a light wrapper around fastlane deliver and transporter app, over time it's become buggy and now it mostly just fails and tells people to use transporter app instead. `upload:ios` also only works on MacOS computers.
 
With the creation of `eas submit` which works on all computers, users now have a couple really good and stable options. We should deprecate `expo upload:ios` and warn users to switch over to EAS. This pattern is similar to other popular React tools like Next.js which uses "next-cli" to start the app, and "now" to build and distribute the app. Removing `upload:ios` gets us closer to pulling Fastlane out of `expo-cli`, which will result in a 30% bundle size decrease, and fix installation issues for many windows and linux users.

# How

Add a notice when running `expo upload:ios`

Screen Shot 2020-12-15 at 2 49 31 PM

Also add a notice to the `--help` prompt

Screen Shot 2020-12-15 at 3 05 13 PM

### CODE REVIEW

**Comment by fson** [comment]
"Deprecated" is different from "removed".
> Deprecated refers to something that should not be used anymore and may be removed in the future.
> But deprecated things are still around and still exist.

So this PR doesn't really "deprecate" the command, but removes it and the wording should be changed. However, to be clear, I fully support this change – `eas submit` is a big improvement over `expo upload:ios` and will provide a much better experience.

**Comment by brentvatne** [comment]
Linking to this comment for more context for people who stumble on this PR: https://github.com/expo/expo-cli/issues/2192#issuecomment-745626708

**Comment by brentvatne** [comment]
Looks good to me, agree with @fson that we should change deprecated to "Removed" or something to that effect

File: packages/expo-cli/src/commands/upload.ts [hunk 1]
```diff
@@ -59,7 +54,9 @@ export default function (program: Command) {
   program
     .command('upload:ios [path]')
     .alias('ui')
-    .description('macOS only: Upload an iOS binary to Apple. An alternative to Transporter.app')
+    .description(
+      `${chalk.yellow('Deprecated:')} Use ${chalk.bold('eas submit')} or Transporter app instead.`
```
Comment (brentvatne) [inline packages/expo-cli/src/commands/upload.ts:58]:
```suggestion
      `${chalk.yellow('Unsupported:')} Use ${chalk.bold('eas submit')} or Transporter app instead.`
```

File: packages/expo-cli/src/commands/upload.ts [hunk 2]
```diff
@@ -101,38 +98,22 @@ export default function (program: Command) {
...
-            TerminalLink.transporterAppLink()
-          )} instead of the ${chalk.bold(
-            'expo upload:ios'
-          )} command if you have any trouble with it.`
-        );
-
-        throw err;
-      }
+      log.newLine();
+      log(chalk.yellow('expo upload:ios is deprecated'));
```
Comment (brentvatne) [inline packages/expo-cli/src/commands/upload.ts:108]:
```suggestion
      log(chalk.yellow('expo upload:ios is no longer supported'));
```

**Review by brentvatne — APPROVED** [review]

**Pull request merged**